azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource.go

```diff
@@ -87,9 +87,10 @@ func resourceArmEventGridEventSubscription() *schema.Resource {
 			},
 
 			"topic_name": {
-				Type:     schema.TypeString,
-				Optional: true,
-				Computed: true,
+				Type:       schema.TypeString,
+				Optional:   true,
+				Computed:   true,
+				Deprecated: "This field has been updated to readonly field since Apr 25, 2019. So remove this unnecessary field.",
```

Comment:
```suggestion
				Deprecated: "This field has been updated to readonly field since Apr 25, 2019 so no longer has any affect and will be removed in version 3.0 of the provider.",
```

Reply:
updated. thx